[front] - refacto(skills): versioning

JulesBelveze opened the pull request:
## Description

Refactors skill versioning to use a dedicated history table instead of a simple counter field. Introduces `SkillVersionModel` to store snapshots of skill configurations before updates, tracking complete historical state including name, description, instructions, status, and associated MCP server configurations. The version counter is removed from `skill_configurations` table as version numbers are now calculated by counting historical records in the new `skill_versions` table.

## Risk

Database schema changes with two migrations (436 and 437). Migration 436 creates the new `skill_versions` table, and migration 437 drops the `version` column from `skill_configurations`. The migrations are safe as the version column was not actively used for querying or business logic - it was only incremented on updates.

## Tests

Locally

## Deploy Plan

1. Deploy front and apply migration 436 (creates `skill_versions` table)
2. Apply migration 437 (drops `version` column from `skill_configurations`)

JulesBelveze committed:
[front/lib/models] - feature: implement skill versioning mechanism

 - Extract common skill model attributes into a constant for reuse and enhanced maintainability
 - Introduce SkillVersionModel to handle different versions of skill configurations
 - Define new associations for SkillVersionModel with UserModel and SkillConfigurationModel
 - Add database indexes to optimize queries on skill versions and ensure data integrity with unique constraints

JulesBelveze committed:
[front/migrations] - feature: add skill_versions table with indexes

 - Created a new table 'skill_versions' to track different versions of skills including metadata and associations
 - Ensured data integrity with foreign keys and cascading behavior upon deletions and updates
 - Added a unique index on skill configuration ID and version to prevent duplicate entries
 - Included an index to improve query performance involving workspace and skill configuration relations

JulesBelveze committed:
[front] - feature: implement versioning for skill configurations

 - Implement a method to save version history for skill configurations before updates
 - Create new version records to track changes to skill configurations in a workspace

JulesBelveze committed:
[front/lib/models/skill] - refactor: remove version field from Skill model

 - Removed the `version` field from the Skill model as it is no longer necessary
 - Updated the `SkillConfigurationResource` class and its methods to function without the `version` attribute

[front/lib/resources/skill] - feature: add method to fetch all active skills

 - Implemented `fetchAllAvailableSkills` to retrieve active skills without the version constraint
 - Adjusted the `updateSkill` and `toJSON` methods to handle the removal of the `version` field

[front/lib/resources/skill] - refactor: reorganize methods in SkillConfigurationResource

 - Moved `fromGlobalSkill` and `saveVersion` to more appropriate places within the class for better readability and organization

JulesBelveze committed:
[db] - feature: remove redundant version column from skill_configurations

 - Dropped the 'version' column from 'skill_configurations' table to streamline database schema as versioning is handled by 'skill_versions' table now

JulesBelveze committed:
[front/types] - refactor: remove version from SkillConfigurationType

 - The `version` field is no longer necessary in skill configuration and has been removed to simplify the structure

JulesBelveze committed:
[front/tests/utils] - refactor: remove version field from SkillConfigurationFactory

 - The `version` field is no longer used in creating skill configurations and has been removed from factory method arguments and object creation.

flvndvd commented on front/lib/models/skill.ts:
```diff
@@ -29,48 +60,49 @@ export class SkillConfigurationModel extends WorkspaceAwareModel<SkillConfigurat
+    skillConfigurationId: {
+      type: DataTypes.BIGINT,
       allowNull: false,
+      references: {
```
We usually materialize this through relationship definition.

flvndvd commented on front/lib/models/skill.ts:
```diff
@@ -29,48 +60,49 @@ export class SkillConfigurationModel extends WorkspaceAwareModel<SkillConfigurat
+      },
+      {
+        unique: true,
+        fields: ["skillConfigurationId", "version"],
```
Let's always have workspace in the index. Also you can probably merge both indexes into one.

flvndvd commented on front/lib/models/skill.ts:
```diff
@@ -150,3 +192,13 @@ SkillMCPServerConfigurationModel.belongsTo(MCPServerViewModel, {
+
+SkillConfigurationModel.hasMany(SkillVersionModel, {
+  foreignKey: { name: "skillConfigurationId", allowNull: false },
+  onDelete: "CASCADE",
```
No way 😛 
```suggestion
  onDelete: "DELETE",
```

flvndvd reviewed (commented):
🔥 🔥 🔥

JulesBelveze committed:
[front/lib/models] - refactor: update skill model attributes and associations

 - Add `Model` and `ModelAttributes` to imports and use `satisfies` for type assertion in `SKILL_MODEL_ATTRIBUTES`
 - Remove explicit foreign key reference in `SkillVersionModel` to `SkillConfigurationModel`
 - Change the deletion policy from `CASCADE` to `RESTRICT` for skill versions tied to configurations

[front/lib/resources] - refactor: remove duplicate type definition for `SkillVersionCreationAttributes`

 - Move `SkillVersionCreationAttributes` type from resource file to model file to consolidate type definitions

[front/migrations] - refactor: alter skill version indexing and delete policy in SQL migration

 - Replace individual indexes with a composite unique index on `workspaceId`, `skillConfigurationId`, and `version`
 - Modify delete policy for `skill_versions` from `CASCADE` to `RESTRICT` to prevent deletion if referenced by another record

JulesBelveze committed:
fix: rebase

JulesBelveze committed:
[front] - test: remove version field in skill API tests

 - The `version` field is no longer expected or required in the skill creation payload during tests

JulesBelveze committed:
[front/lib/resources/skill] - refactor: remove computeCanEdit from SkillResource

 - Deleted the method computeCanEdit as its functionality is possibly offloaded elsewhere or no longer needed
 - Simplification of the SkillResource code by removing unnecessary user permissions logic
